models/login_source.go

```diff
@@ -361,10 +377,13 @@ func (s *LocalLoginSources) ActivatedList() []*LoginSource {
 		if !s.sources[i].IsActived {
 			continue
 		}
-
 		source := &LoginSource{}
 		*source = *s.sources[i]
-		list = append(list, source)
+		if s.sources[i].IsDefault {
+			list = append([]*LoginSource{source}, list...)
+		} else {
+			list = append(list, source)
+		}
```

Comment:
What will be the effect if we do not make the "default" login source as the first element in the slice?

Reply:
When responding to the client, the default will be made, so no need to care about order here.

https://github.com/gogs/gogs/pull/5338/files#diff-d34865bc980845207ee9456ffb644282R116